en/scripting/modules/spec.md

```diff
@@ -0,0 +1,195 @@
+
+### Bare Specifiers
+
+For bare specifiers, Cocos Creator will apply the Node.js module parsing algorithm.
+
+> This includes parsing of npm modules.
+
+In general, bare specifiers have the following two forms:
+
+- `'foo'`: parsed as the entry module for the npm package `foo`.
+
+- `'foo/bar'`: parsed as the module under the subpath `./bar` in the npm package `'foo'`.
+
+The specific rules for parsing bare specifiers can be found in [Node.js module parsing algorithm](https://nodejs.org/api/esm.html#esm_resolver_algorithm_specification).
+
+> In the future, Cocos Creator may support importing maps, see [Import Maps](https://github.com/WICG/import-maps).
+
+### Suffixes and Directory Import
+
+Cocos Creator's requirements for module suffixes in module specifiers are more web-oriented -- suffixes must be specified and Node.js-style directory import is not supported. However, for historical reasons and some existing restrictions, TypeScript modules do not allow suffixes and support Node.js-style directory import. Specifically:
+
+When the target module file has the suffix `.js`, `.mjs`, the suffix **must be specified** in the module specifier: `.js`, `.mjs`.
+
+```ts
+import '. /foo.mjs'; // correct
+import '. /foo'; // error: the specified module cannot be found
+```
+
+Node.js-style directory import is not supported:
+
+```ts
+import '. /foo/index.mjs'; // correct
+import '. /foo'; // error: module cannot be found.
+```
+
+> This suffix requirement applies to both relative and absolute specifiers along with the restriction on directory import. For requirements in bare specifiers please refer to the Node.js module parsing algorithm.
+
+However, when the target module file has a suffix of `.ts`, the suffix is **not allowed to be specified** in the module specifier:
+
+```ts
+import '. /foo'; // correct: parsed as the `foo.ts` module in the same directory
+import '. /foo.ts'; // error: the specified module cannot be found
+```
+
+On the other hand, Node.js-style directory import is supported:
+
+```ts
+import '. /foo'; // correct: parsed as the `foo/index.ts` module
+```
+> **Note**:
+> Cocos Creator supports the Web platform. Implementing complex module parsing algorithms like Node.js on the Web platform is expensive, and the client and server cannot try different suffixes and file paths with frequent communication between them.
+> Even if such complex parsing could be done at the build stage with some post-processing tools, it would result in inconsistent algorithms for static import parsing (via `import` statements) and dynamic import parsing (via `import()` expressions). Therefore, specify the full file path in the code for the choice of module parsing algorithm.
```

Comment:
```suggestion
> 2. Even if such complex parsing could be done at the build stage with some post-processing tools, it would result in inconsistent algorithms for static import parsing (via `import` statements) and dynamic import parsing (via `import()` expressions). Therefore, specify the full file path in the code for the choice of module parsing algorithm.
```